### PR TITLE
Feature/18216/zen mode

### PR DESCRIPTION
**Pull request by wielinde**
https://community.openproject.com/projects/openproject/work_packages/18216/activity

### CODE REVIEW

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/c5d0b49159aab6bfdf050d76d53f729babb05963) (deployed commit: ce8f6d1bac6ee5e7e626459f2ea8a0291297f0ae).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/8730c0d411ac978126ac97425d4cd1191965e78e) (deployed commit: 9f797d80baaf6562aa648f7a9357c86d16e00824).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/79060e30b94e0fb7d4cec3e85cde55d48992c4bc) (deployed commit: 5e85b987c6ab0ba9fbda03cef0a697105e794f28).

**Comment by lindenthal** [comment]
@wielinde the new zen mode feels great!

I justed tested it in Edge, FF and Chrome on Windows 10. I didn't find any bugs.

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/09b2972aaf762854195da2c56d5759ecd98c0e35) (deployed commit: 20e8dd1bb9ab72a2cbfa049a0b3dac05f1e1efab).

**Review by oliverguenther — CHANGES_REQUESTED** [review]
Please avoid vendoring libraries whenever possible, instead use their npm package counterpart. There appears to be none for the plugin you're using, however there are others that appear to be more active:

https://github.com/private-face/jquery.fullscreen
https://www.npmjs.com/package/jquery.fullscreen

This avoids us having to maintain and manually upgrade the vendored code.
Otherwise, the feature appears to be working fine 👍  Good to merge after travis agrees and ideally, you replaced the library with a package.

File: app/assets/javascripts/application.js.erb [hunk 1]
```diff
@@ -35,6 +35,7 @@
 //= require lib/jquery.colorcontrast
 //= require lib/jquery.trap
 //= require lib/jquery.is_visible_within
+//= require lib/jquery.fullscreen
```
Comment (oliverguenther) [inline app/assets/javascripts/application.js.erb]: Instead requiring it through sprockets, install the package in the frontend folder witwh `npm install --save <package` and `require('package');` it in `frontend/app/vendors.js`

**Comment by wielinde** [comment]
@oliverguenther Thank you for the review. I now changed the code to use the npm package *screenfull*.

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/88bdd5f2a183abb8e8bc5ad255247176178a69e6) (deployed commit: b48a51a5a23e4c3ed3e0dc71ece61e12f0d96c27).

**Review by oliverguenther — APPROVED** [review]

**Comment by oliverguenther** [comment]
Nice! the failing test is unrelated and caused by a order dependent failure.

**Pull request merged**